### PR TITLE
[APB-4379][SM]: Known fact check for GB clients: postcode

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/agentinvitationsfrontend/models/Invitation.scala [hunk 1]
```diff
@@ -16,47 +16,11 @@
...
-object UTR {
-  implicit val format: Format[UTR] = Json.format[UTR]
-}
-
-case class CountryCode(value: String) extends KnownFact
-
-object CountryCode {
-  implicit val format: Format[CountryCode] =
-    implicitly[Format[String]].inmap(CountryCode(_), _.value)
-}
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/agentinvitationsfrontend/models/Invitation.scala]: remove all above knownFacts  as we are not using them apart from setting them

File: app/uk/gov/hmrc/agentinvitationsfrontend/models/Postcode.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package uk.gov.hmrc.agentinvitationsfrontend.models
+
+import play.api.libs.json.{Format, Json}
+
+case class Postcode(value: String)
+
+object Postcode {
+  implicit val format: Format[Postcode] = Json.format[Postcode]
```
Comment (stevenmallen) [inline app/uk/gov/hmrc/agentinvitationsfrontend/models/Postcode.scala]: Please remove redundant type parameter from rhs

File: conf/messages [hunk 3]
```diff
@@ -126,6 +126,10 @@ identify-cgt-client.trust.header.business=What is the trust''s Capital Gains Tax
 identify-cgt-client.p1=This will help us match their details against information we hold.
 identify-cgt-client.p2=Capital Gains Tax (CGT) account reference
 identify-cgt-client.hint=This is 15 characters, for example XYCGTP123456789.  Your client received this when they created their CGT account.
+confirm-postcode-cgt.header=What is your client''s postcode?
```
Comment (stevenmallen) [inline conf/messages]: Please use correct unicode quotes here (U+2019) - this is required by HMRC.

File: it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationJourneyControllerISpec.scala [hunk 4]
```diff
@@ -546,7 +546,7 @@ class AgentInvitationJourneyControllerISpec extends BaseISpec with StateAndBread
             case ConfirmClientItsa(
                 AuthorisationRequest(
                   "Sylvia Plath",
-                  ItsaInvitation(_, Postcode("BN114AW"), _, _, _),
+                  ItsaInvitation(_, _, _, _),
```
Comment (stevenmallen) [inline it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationJourneyControllerISpec.scala]: Why was the postcode check removed?

Comment (sureshhmrc) [reply it/uk/gov/hmrc/agentinvitationsfrontend/controllers/AgentInvitationJourneyControllerISpec.scala]: because we are not using that (knownfact)

**Pull request merged**